ADR 154 Multiplayer out of the box (#155)
decentraland/adr: content/ADR-154-multiplayer-ootb.md

---
layout: adr
adr: 154 # replace this number for the PR or ISSUE number
title: Multiplayer out of the box
date: 2022-12-15
status: Idea # pick one of these
type: RFC # pick one of these
spdx-license: CC0-1.0
authors:
  - nearnsahw # this is your github username
# remove the following line! it exists to render the template nicely
---

## Abstract

<!--
Abstract is a multi-sentence (short paragraph) technical summary. This should be a very terse and human-readable version of the document section. **Someone should be able to read only the abstract to get the gist of what this document is about in its current state.** Abstracts should be always up to date with the current state of the document.
-->

> The purpose of this feature is to ,ake it easier for content creators to build scenes where what one player changes is seen and shared by all other players there. This today is possible but very hard, and because of that very few creators explore those options.

Reviewer: ```suggestion
The purpose of this feature is to make it easier for content creators to build scenes where what one player changes is seen and shared by all other players there. This today is possible but very hard, and because of that very few creators explore those options.
```

The exact details about how this feature is to be implemented are yet to be defined.

## Context, Reach & Prioritization

<!--
Discuss and go into detail about the subject in question. Make sure you cover:
- Why is this decision important
- The urgency of the decision
- Datapoints and related background information
- Vocabulary and key terms
-->

There are currently very few scenes where what one player does affects another. This results in very few truly social shared experiences. Mostly each player is in a single-player experience, watching other avatars walking around in their own separate dimension. This is confusing to new players, and severely limits what can be done with Decentraland as a social space. This is not just about making multiplayer games easier, the environment and its changes can be used a means for socializing, and can enhance every kind of experience.

> We believe few people create these kinds of experiences because it’s hard. Our SDK doesn’t offer any way to simplify the syncing of state between players. Content creators have to either set up a whole server themselves, or manually send messages and then interpret them with custom logic. This can also result in a buggy experience as messages could arrive out of order and different player’s chances can enter into conflict.

Reviewer: ```suggestion
We believe few people create these kinds of experiences because it's hard. The SDK doesn’t offer any way to simplify the syncing of state between players. Content creators have to either set up a whole server themselves, or manually send messages and then interpret them with custom logic. This can also result in a buggy experience as messages could arrive out of order and different player's chances can enter into conflict.
```

## Solution Space Exploration

<!--
Discuss the potential alternatives and their impact. What alternatives are being considered, their benefits, their costs (team resources, money, time frames), and mitigations for any drawbacks.
-->

> The proposed solution involves allowing content creators to easily mark an entity or a component as synced. Once that’s done, if any properties change in a marked component or entity, these changes are shared with all other players who are sharing an island.

Reviewer: ```suggestion
The proposed solution involves allowing content creators to easily mark an entity or a component as synced. Once that's done, if any properties change in a marked component or entity, these changes are shared with all other players who are sharing an island.
```

---
layout: adr
adr: 154 # replace this number for the PR or ISSUE number
title: Multiplayer out of the box
date: 2022-12-15
status: Idea # pick one of these
type: RFC # pick one of these
spdx-license: CC0-1.0
authors:
  - nearnsahw # this is your github username
# remove the following line! it exists to render the template nicely

> slug: /adr/multiplayer-out-of-the-box

Reviewer: ```suggestion
```
